### ISSUE TITLE
Search resource implementation

### ISSUE THREAD

**Issue by carlespibernat** [api-platform/core#1995]
Hello!

I want to create a searcher in my project which searches and gives results of different resources. 

I suppose that the most "RESTful" way to do that is to create a search resource, so what I have done is create my own `Search` entity. Then i have created a `SearchItemDataProvider` that will get the filters from the request and will find the results and return them. 

The problem that I have now is that api platform requires that my resource has a specific id for each search, so I get an error.

Is this the properly way to implement this? And if it is how can I solve my issue with the id?

Thanks!

**Comment by teohhanhui** [comment]
> What I'd do in your case would be to implement a custom Search filter (extends the Abstractfilter). Then you declare the filter on each resource you want to filter.

No, that wouldn't work as what's being asked for is a search resource that aggregates other resources.

**Comment by teohhanhui** [comment]
I think a major blocker for this kind of use case is the lack of support for blank nodes: https://json-ld.org/spec/latest/json-ld/#identifying-blank-nodes

**Comment by carlespibernat** [comment]
> No, that wouldn't work as what's being asked for is a search resource that aggregates other resources.

You're right @teohhanhui.

What I said was inspired by [this](https://stackoverflow.com/a/18933902/9149444) stackoverflow post answer.

**Comment by teohhanhui** [comment]
Yes, I think that's a good pattern. We should support it.

**Comment by soyuka** [comment]
> No, that wouldn't work as what's being asked for is a search resource that aggregates other resources.

Okay I misunderstood the question then :D. 
@teohhanhui we already have an issue open for blank nodes no? Maybe time to dig into this...

**Comment by carlespibernat** [comment]
Then it's not possible to do this for the moment?

**Comment by teohhanhui** [comment]
@carlespibernat I don't think it's impossible, but it's harder.
